apps/pollbook/backend/src/pollbook_package.ts

```diff
@@ -255,3 +258,100 @@ export function pollUsbDriveForPollbookPackage({
     }, 100);
   });
 }
+
+export function pollNetworkForPollbookPackage({
+  auth,
+  workspace,
+}: PeerAppContext): void {
+  usbDebug('Polling network for pollbook package');
+  if (workspace.store.getElection()) {
+    return;
+  }
+  let pollingIntervalLock = false; // Flag to prevent overlapping executions
+  let hadConfigurationError = false;
+
+  process.nextTick(() => {
+    const intervalId = setInterval(async () => {
+      if (pollingIntervalLock) {
+        return; // Skip if a polling iteration is already in progress
+      }
+      pollingIntervalLock = true; // Set the flag to indicate polling is in progress
+
+      try {
+        if (workspace.store.getElection()) {
+          clearInterval(intervalId); // Stop the polling interval
+          return;
+        }
+
+        const authStatus = await auth.getAuthStatus(
+          constructAuthMachineState(workspace)
+        );
+        if (!isElectionManagerAuth(authStatus)) {
+          workspace.store.setConfigurationStatus(undefined);
+          usbDebug('Not logged in as election manager, not configuring');
+          hadConfigurationError = false;
+          return;
+        }
+        if (hadConfigurationError) {
+          return;
+        }
+        const cardElectionId = authStatus.user.electionKey.id;
+        const pollbooksOnNetwork = workspace.store.getPollbookServicesByName();
+        const configuredPollbooks = Object.values(pollbooksOnNetwork).filter(
+          (pollbook) =>
+            pollbook.status === PollbookConnectionStatus.WrongElection &&
+            pollbook.electionId
+        );
+        const matchingConfiguredPollbooks = configuredPollbooks.filter(
+          (p) => p.electionId === cardElectionId
+        );
+        const allPollbooksHashesMatch =
+          matchingConfiguredPollbooks.length > 0 &&
+          matchingConfiguredPollbooks.every(
+            (pollbook) =>
+              pollbook.electionBallotHash ===
+                matchingConfiguredPollbooks[0].electionBallotHash &&
+              pollbook.pollbookPackageHash ===
+                matchingConfiguredPollbooks[0].pollbookPackageHash
+          );
+
+        if (configuredPollbooks.length === 0) {
+          workspace.store.setConfigurationStatus('not-found-network');
+          return;
+        }
```

Comment:
nit: do this check on L305, closer to assignment and avoiding unnecessary list operations?